### PR TITLE
added conda scraper

### LINKED ISSUE #31: Generate conda dataset

### LINKED ISSUE #33: Create Conda Scraper for conda-forge packages
Scraping data from conda site for python packages.

### PR DESCRIPTION
**Pull request by srossross**
@siddheshghadi751 this supersedes  #37

### CODE REVIEW

File: score/cli.py [hunk 1]
```diff
@@ -110,5 +112,46 @@ def scrape_pypi_web(num_partitions, partition, output):
...
+    "--output",
+    default=OUTPUT_ROOT / "output" / "conda",
+    help="The output directory to save the scraped data in hive partition",
+)
+@click.option(
+    "-c",
+    "--channel",
+    default="conda-forge",
+    help="The conda channel to scrape packages from",
+)
```
Comment (srossross) [inline score/cli.py:126]: channel is part of the cli options

File: score/conda/get_conda_package_names.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from typing import List
+from ..utils.request_session import get_session
+from hashlib import sha256
+
+CONDA_URL_TEMPLATE = "https://conda.anaconda.org/{channel}/linux-64/repodata.json"
```
Comment (srossross) [inline score/conda/get_conda_package_names.py:5]: use the this url for list of package names, only the conda-forge channel has a 'feedstock' url

File: score/conda/scrape_conda.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        source_url = package_data.get("dev_url") or package_data.get("source_git_url")
+        all_packages.append(
+            {
+                "name": package_data["name"],
+                "full_name": package_data["full_name"],
+                "source_url": source_url,
+                "latest_version": package_data["latest_version"],
+                "ndownloads": ndownloads,
+            }
+        )
```
Comment (srossross) [inline score/conda/scrape_conda.py:31]: only relevant data we have identified is needed

File: score/conda/scrape_conda.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+from ..utils.request_session import get_session
+
+CONDA_PACKAGE_URL_TEMPLATE = "https://api.anaconda.org/package/{channel}/{package}"
+
+
+def scrape_conda(channel, package_names: List[str]) -> pd.DataFrame:
+    s = get_session()
+    all_packages = []
+    for package in tqdm(package_names):
```
Comment (srossross) [inline score/conda/scrape_conda.py:13]: use tqdm

File: score/cli.py [hunk 5]
```diff
@@ -110,5 +112,46 @@ def scrape_pypi_web(num_partitions, partition, output):
...
+    type=int,
+    help="The partition number to process.",
+)
+def conda(num_partitions, partition, output, channel):
+    packages = get_conda_package_names(num_partitions, partition, channel)
+    click.echo(
+        f"Will process {len(packages)} packages in partition {partition} of {num_partitions}"
+    )
+
+    df = scrape_conda(channel, packages)
```
Comment (srossross) [inline score/cli.py:147]: the scraper does not know about the output or partitions it only gets data for what is requested

File: score/cli.py [hunk 6]
```diff
@@ -110,5 +112,46 @@ def scrape_pypi_web(num_partitions, partition, output):
...
+    df["partition"] = partition
+    df["channel"] = channel
+
+    click.echo(f"Saving data to {output}")
+    df.to_parquet(output, partition_cols=["channel", "partition"])
```
Comment (srossross) [inline score/cli.py:152]: partition based on "channel" and "partition" columns

File: score/conda/scrape_conda.py (hunk 3 above)
Comment (srossross) [reply score/conda/scrape_conda.py:31]: @karamba228 we should normalize the source url some have a trailing `/`

Comment (srossross) [reply score/conda/scrape_conda.py:31]: issue #50

File: score/conda/get_conda_package_names.py (hunk 2 above)
Comment (siddheshghadi751) [reply score/conda/get_conda_package_names.py:5]: but on browser when I pass `conda-forge` as channel and doesn't show any data when it loads

**Review by siddheshghadi751 — APPROVED** [review]

**Pull request merged**